main/ble_utils.c

```diff
@@ -873,22 +873,29 @@ ble_characteristic_t *ble_device_characteristic_add(ble_service_t *service,
     characteristic->handle = handle;
     characteristic->properties = properties;
     characteristic->client_config_handle = 0;
+    characteristic->index = 0;
+
+    for (cur = &service->characteristics; *cur; cur = &(*cur)->next){
+        if (ble_uuid_equal((*cur)->uuid, uuid)){
+            characteristic->index++;
+        }
+    };
```

Comment:
Why are you iterating over the service's characteristics again to find the index? It should be passed to this function as an argument.
Why are you opposed to the changes I suggested in `ble_update_cache()` to calculate the index while adding the characteristics?

Reply:
we will need to iterate over the existing characteristics in any case, since the repeating UUID dont have to follow each other directly.

Reply:
Understood, if it's not guaranteed then we have to count them each time. Still, I think it should be in `ble_update_cache()` and not in the utils.

Reply:
like it is now?